jetty-core/jetty-fcgi/jetty-fcgi-server/src/main/java/org/eclipse/jetty/fcgi/server/internal/ServerFCGIConnection.java

```diff
@@ -407,4 +408,12 @@ public void onFailure(int request, Throwable failure)
             stream = null;
         }
     }
+
+    @Override
+    public void close()
+    {
+        if (stream != null)
+            stream.getHttpChannel().onFailure(new EofException());
```

Comment:
`onFailure()` returns a `Runnable` that must be run.

Reply:
Done.